vizro-core/src/vizro/models/_components/form/checklist.py

```diff
@@ -46,9 +48,12 @@ class Checklist(VizroBaseModel):
     _validate_options = root_validator(allow_reuse=True, pre=True)(validate_options_dict)
     _validate_value = validator("value", allow_reuse=True, always=True)(validate_value)
 
-    @_log_call
-    def build(self):
-        full_options, default_value = get_options_and_default(options=self.options, multi=True)
+    def __call__(self, new_options=None, **kwargs):
+        return self._build_static(new_options=new_options, **kwargs)
+
+    def _build_static(self, new_options=None, **kwargs):
+        options = new_options if new_options else self.options
```

Comment:
Unless there's good reason to not do it this way let's make these arguments non-optional and just supply them all the time. Hiding the defaulting behaviour inside the `_build_static` function is more confusing I think.
```suggestion
    def _build_static(self, options):
```

Similar comment across all selectors.

Reply:
Looks good, thanks!